src/simulation/deviceInterface/tempMeasurement/tempMeasurement.rst

```diff
@@ -0,0 +1,111 @@
+Executive Summary
+-----------------
+Models the addition of noise, bias, and faults to temperature measurements.
+
+Message Connection Descriptions
+-------------------------------
+The following table lists all module input and output messages. The module msg connection 
+is set by the user from python. The msg type contains a link to the message structure definition, 
+while the description provides information on what this message is used for.
+
+.. list-table:: Module I/O Messages
+    :widths: 25 25 50
+    :header-rows: 1
+
+    * - Msg Variable Name
+      - Msg Type
+      - Description
+    * - tempInMsg
+      - :ref:`TemperatureMsgPayload`
+      - True temperature measurement
+    * - tempOutMsg
+      - :ref:`TemperatureMsgPayload`
+      - Sensed temperature measurement with corruptions
+
+Detailed Module Description
+---------------------------
+
+This module simulates the corruption of a true thermal measurement by noise,
+bias, and three faults:
+
+- "TEMP_FAULT_STUCK_VALUE" is faulty behavior where the measurement sticks to a specific value
+- "TEMP_FAULT_STUCK_CURRENT" fixes the measurement to the value
+- "TEMP_FAULT_SPIKING" is faulty behavior where the measurement spikes 
+  to a specified multiplier times the actual value, with a given probability
+- "TEMP_FAULT_NOMINAL" has no faulty behavior but may still have noise and bias
+
+
+User Guide
+----------
+
+Fault Parameters
+~~~~~~~~~~~~~~~~
+
+This module has several parameters that are set to default values:
+
+.. list-table:: Default Module Parameters
+    :widths: 25 50 25
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Default Value
+    * - faultState
+      - Sets the fault status.
+      - TEMP_FAULT_NOMINAL
+    * - senBias
+      - Sets the bias value.
+      - 0.0
+    * - senNoiseStd
+      - Sets the standard deviation for sensor noise.
+      - 0.0
+    * - walkBounds
+      - Sets the random walk bounds for sensor noise.
+      - 1E-15
+    * - stuckValue
+      - Temperature at which the reading is stuck for fault mode TEMP_FAULT_STUCK_VALUE.
+      - 0.0
+    * - spikeProbability
+      - Probability of a spike when in fault mode TEMP_FAULT_SPIKING. Between 0 and 1.
```

Comment:
use `` to mark TEMP_FAULT_SPIKING as a variable

Reply:
Done.